test/freenet/client/filter/OggFilterTest.java

```diff
@@ -24,49 +22,60 @@ public void setUp() {
+			assertThrows(
+				DataFilterException.class,
+				() -> filter.readFilter(input, new ByteArrayOutputStream(), null, null, null, null)
+			);
+		});
 	}
 
 	@Test
-	public void testValidSubPageStripped() throws IOException, DataFilterException {
-		DataInputStream input = new DataInputStream(getClass().getResourceAsStream("./ogg/contains_subpages.ogg"));
-		ByteArrayOutputStream output = new ByteArrayOutputStream();
-		try {
-			filter.readFilter(input, output, null, null, null, null);
-		} catch(DataFilterException e) {}
-		assertTrue(Arrays.equals(new byte[]{}, output.toByteArray()));
-		input.close();
-		output.close();
+	public void testValidSubPageStripped() throws IOException {
+		testResourceFile("./ogg/contains_subpages.ogg", (input) -> {
+			try (ByteArrayOutputStream output = new ByteArrayOutputStream()) {
+				assertThrows(
+					DataFilterException.class,
+					() -> filter.readFilter(input, output, null, null, null, null)
+				);
+				assertArrayEquals(new byte[]{}, output.toByteArray());
+			}
+		});
 	}
 
     /** the purpose of this test is to create the testoutputFile so you can check it with a video player. */
 	@Test
-	public void testFilterFfmpegEncodedVideoSegment() throws IOException, DataFilterException {
-        String testoutputFile = getClass().getResource(
-	 			"./ogg/36C3_-_opening--cc-by--c3voc--fem-ags-opensuse--ccc--filtered-testoutput.ogv")
-					.getFile();
-		DataInputStream inputFileUnchanged = new DataInputStream(getClass().getResourceAsStream(
-				"./ogg/36C3_-_opening--cc-by--c3voc--fem-ags-opensuse--ccc--filtered.ogv"));
-		ByteArrayOutputStream unchangedData = new ByteArrayOutputStream();
-		IOUtils.copy(inputFileUnchanged, unchangedData);
-		DataInputStream inputFileToParse = new DataInputStream(getClass().getResourceAsStream(
-				"./ogg/36C3_-_opening--cc-by--c3voc--fem-ags-opensuse--ccc--orig.ogv"));
-		DataInputStream input = inputFileToParse;
-		ByteArrayOutputStream output = new ByteArrayOutputStream();
-		try {
-			filter.readFilter(input, output, null, null, null, null);
-			FileOutputStream newFileStream = new FileOutputStream(testoutputFile);
-			System.out.println(testoutputFile);
+	public void testFilterFfmpegEncodedVideoSegment() throws IOException {
+		try (
+			ByteArrayOutputStream unchangedData = new ByteArrayOutputStream();
+			ByteArrayOutputStream output = new ByteArrayOutputStream()
+		) {
+			testResourceFile(
+				"./ogg/36C3_-_opening--cc-by--c3voc--fem-ags-opensuse--ccc--filtered.ogv",
+				(inputFileUnchanged) -> IOUtils.copy(inputFileUnchanged, unchangedData)
+			);
+			testResourceFile(
+				"./ogg/36C3_-_opening--cc-by--c3voc--fem-ags-opensuse--ccc--orig.ogv",
+				(input) -> {
+					filter.readFilter(input, output, null, null, null, null);
+					writeToTestOutputFile(output);
+				}
+			);
+			assertArrayEquals(unchangedData.toByteArray(), output.toByteArray());
+		}
+	}
+
+	private void writeToTestOutputFile(ByteArrayOutputStream output) throws IOException {
```

Comment:
Oof, what is even this?! Looks like somebody committed debugging code! (Again, no action needed.)